clio-integration-testkit/src/main/scala/org/broadinstitute/clio/integrationtest/ClioDockerComposeContainer.scala

```diff
@@ -0,0 +1,331 @@
+          }
+
+          private val lineMarkersToPrint = Seq("INFO", "WARN", "ERROR")
+
+          override def handle(line: String): Unit = {
+            if (lineMarkersToPrint.exists(line.contains)) {
+              println(line)
+            }
+            if (line.contains(readyLog)) {
+              tailer.stop()
+            }
+          }
+        }
+
+        val tailer = new Tailer(clioLog.toJava, listener, 250)
+
+        println("Waiting for Clio to log startup message...")
+        tailer.run()
+        println("Clio ready!")
+      }
+    }.withStartupTimeout(Duration.ofSeconds(90)))
+
+  lazy val container = DockerComposeContainer(
+    ClioDockerComposeContainer.extractComposeFiles(tmpDir).toJava,
+    exposedServices = Seq(
+      clioService,
+      ClioDockerComposeContainer.esService
+    ),
+    /*
+     * Skip "docker pull" because it'll fail if we're testing against a non-
+     * pushed version of clio-server.
+     */
+    pull = false,
+    /*
+     * Testcontainers doesn't pass env vars onto the docker-compose container it
+     * spins up, and we need the vars in order to fill in our compose config.
+     * Using the local install gets the vars to pass through.
+     */
+    localCompose = true,
+    /*
+     * Set environment variables to ensure we're testing the right versions,
+     * persisting files to local disk, and avoiding timezone drift.
+     */
+    env = Map(
+      ClioDockerComposeContainer.clioVersionVariable -> TestkitBuildInfo.version,
+      ClioDockerComposeContainer.elasticsearchVersionVariable -> TestkitBuildInfo.elasticsearchVersion,
+      ClioDockerComposeContainer.logDirVariable -> logDir.toString,
+      ClioDockerComposeContainer.clioLogFileVariable -> clioLog.toString,
+      ClioDockerComposeContainer.persistenceDirVariable -> persistenceDir.toString,
+      ClioDockerComposeContainer.timezoneVariable -> TimeZone.getDefault.getID,
+      ClioDockerComposeContainer.umaskVariable -> "0000"
+    )
+  )
+
+  /**
+    * Make sure log files exist with global rw permissions before starting containers.
+    *
+    * Global rw permissions are needed so the integration tests can run on Jenkins,
+    * which uses an older version of Docker that bind-mounts files into containers
+    * with weird uids and gids. If permissions aren't set properly, log4j will fail
+    * to initialize in the ES containers and tests will fail to start.
+    */
+  override def starting()(implicit description: Description): Unit = {
+
+    // Clean out the log directory so the Tailer won't exit early from examining previous logs.
+    if (logDir.exists) {
+      logDir.delete()
+    }
+    logDir.createDirectories()
+
+    // Extract logging config to mount.
+    val _ = (
+      ClioDockerComposeContainer.extractClioLogback(logDir),
+      ClioDockerComposeContainer.extractEsLog4j(logDir)
+    )
+
+    // Make sure target log files are writeable.
+    (clioLog +: esLogs).foreach { log =>
```

Comment:
Why do we need to do this?

Reply:
If the files don't exist, Docker mounts the paths as directories and the servers fall over trying to write logs. If the files exist but aren't writeable, Docker doesn't chmod them when it mounts so the servers still fall over trying to write.